pkg/schedule/diagnostic/diagnostic_manager.go

```diff
@@ -0,0 +1,79 @@
+// Copyright 2022 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package diagnostic
+
+import (
+	"time"
+
+	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/pkg/schedule/schedulers"
+	"github.com/tikv/pd/server/config"
+)
+
+// Manager is used to manage the diagnostic result of schedulers for now.
+type Manager struct {
+	config              *config.PersistOptions
+	schedulerController map[string]*schedulers.ScheduleController
+}
+
+// NewManager creates a new Manager.
+func NewManager(schedulerController map[string]*schedulers.ScheduleController, config *config.PersistOptions) *Manager {
+	return &Manager{
+		config:              config,
+		schedulerController: schedulerController,
+	}
+}
+
+// GetDiagnosticResult gets the diagnostic result of the scheduler.
+func (d *Manager) GetDiagnosticResult(name string) (*schedulers.DiagnosticResult, error) {
+	if !d.config.IsDiagnosticAllowed() {
+		return nil, errs.ErrDiagnosticDisabled
+	}
+
+	scheduler, isSchedulerExisted := d.schedulerController[name]
```

Comment:
I think there's going to be a data race. We handle schedulers in coordinator with lock, but here is no lock.

Reply:
Yes, add a lock, and I will remove it in the next PR.